x/gamm/keeper/pool_service_test.go

```diff
@@ -203,241 +151,347 @@ func (suite *KeeperTestSuite) TestCreateBalancerPool() {
 			}, {
 				Weight: sdk.NewInt(100),
 				Token:  sdk.NewCoin("foo", sdk.NewInt(10000)),
-			}}, defaultFutureGovernor)
-			_, err := keeper.CreatePool(suite.Ctx, msg)
-			suite.Require().Error(err, "can't create the pool with duplicated PoolAssets")
+			}}, defaultFutureGovernor),
+			expectPass: false,
 		},
-	}, {
-		fn: func() {
-			keeper := suite.App.GAMMKeeper
-			keeper.SetParams(suite.Ctx, types.Params{
-				PoolCreationFee: sdk.Coins{},
-			})
-			msg := balancer.NewMsgCreateBalancerPool(suite.TestAccs[0], balancer.PoolParams{
+	}
+
+	for _, test := range tests {
+		suite.SetupTest()
+		keeper := suite.App.GAMMKeeper
+
+		// fund sender test account
+		sender, err := sdk.AccAddressFromBech32(test.msg.Sender)
+		suite.Require().NoError(err, "test: %v", test.name)
+		if sender.String() != emptySender.String() {
+			suite.FundAcc(sender, defaultAcctFunds)
+		}
+
+		// note starting balances for community fee pool and pool creator account
+		feePoolBeforeNewPool := suite.App.DistrKeeper.GetFeePoolCommunityCoins(suite.Ctx)
```

Comment:
nit: consider changing to:

```suggestion
		feePoolBalBeforeNewPool := suite.App.DistrKeeper.GetFeePoolCommunityCoins(suite.Ctx)
```

Reply:
The original version of this was just `feePool` so I felt like `feePoolBeforeNewPool` was already pretty verbose. If you feel like this would have made it more readable for you then definitely happy to change

Reply:
yeah would prefer changing to keep consistency with `senderBalBefore`! Thnx!

Reply:
sounds good, just changed! (couldn't commit your suggestion directly since it would break things hope that's okay)